doc/cephfs/client-auth.rst

```diff
@@ -153,3 +153,45 @@ The optional ``{network/prefix}`` is a standard network name and
 prefix length in CIDR notation (e.g., ``10.3.0.0/16``).  If present,
 the use of this capability is restricted to clients connecting from
 this network.
+
+Filesystem information restriction
+==================================
+
+If desired, the monitor cluster can present a limited view of the filesystems
+available. In this case, the monitor cluster will only inform clients about
+filesystems specified by the administrator. Other filesystems will not be
+reported and commands affecting them will be ignored as if the filesystems
+did not exist.
+
+::
+
+ client.0
+     key: AQAz7EVWygILFRAAdIcuJ12opU/JKyfFmxhuaw==
+     caps: [mon] allow r, allow fsid=1, allow fsid=2
```

Comment:
These caps are additive. So: you're basically saying the client can do anything with the monitors allowed by `r`, also `r` for `allow fsid=1`, and `r` again for `allow fsid=2`. So the latter two clauses have no effect.

This should be:

`caps: [mon] allow r fsid=1, allow r fsid=2`

Reply:
Yep, I am working on it.

Reply:
This commit's message,
```
Add a 'fsid' clause to mon auth caps to restrict a client's view
of the FSMap. Example:

mon 'allow rw; allow fsid 2'

This would restrict the client's view of the FSMap to the MDSMap for
fscid 2. 
```
It's incorrect, right? Should be mon 'allow rw fsid=2'

Reply:
> beside the document change. Does the code correctly enforce caps: [mon] allow r fsid =1

It works fine AFAIR. I haven't tested it as extensively as the MDS caps.

> It's incorrect, right? Should be mon 'allow rw fsid=2'

Yes, it should be "allow rw fsid=2". And when multiple caps are required it should be "allow rw fsid=1, allow rw fsid=2". The documentation is not updated yet. I intended to update docs after I get an approval regarding current syntax of caps from reviewers.

Reply:
It works -

```
$ ./bin/ceph --id someuser4 -k ceph.client.someuser4.keyring fs dump -f json-pretty | grep \"id\"
dumped fsmap epoch 9
            "id": 2
$ cat ceph.client.someuser4.keyring
[client.someuser4]
	key = AQB6p31e9wdMMRAAlP58QpKut7ALAW02WVmXYw==
	caps mds = "allow rw fsid=2"
	caps mon = "allow rw fsid=2"
	caps osd = "allow rw tag cephfs data=b"
```